net-p2p/amule/files/amule-2.3.3-backport-pr368.patch

```diff
@@ -0,0 +1,44 @@
+https://bugs.gentoo.org/841296
+https://github.com/amule-project/amule/issues/342
+https://github.com/amule-project/amule/pull/368
+
+From dcb419305abbe3defbfa4b18ec0d8a2e76041b04 Mon Sep 17 00:00:00 2001
+From: matoro <matoro@users.noreply.github.com>
+Date: Thu, 14 Dec 2023 00:28:50 -0500
+Subject: [PATCH] SafeFile: do endian-swap on float tag values
+
+There are comments noting that writing raw float values may not be
+endian-correct and indeed it is not.  WX does not provide functions for
+endian-swapping floats, but since amule explicitly uses exclusively
+single-precision (32-bit) floats, just cast it into a uint32_t for the
+swapping.
+
+This is a no-op on little-endian.
+---
+ src/SafeFile.cpp | 6 +++++-
+ 1 file changed, 5 insertions(+), 1 deletion(-)
+
+diff --git a/src/SafeFile.cpp b/src/SafeFile.cpp
+index 30f9a7f23..6e9b07df5 100644
+--- a/src/SafeFile.cpp
++++ b/src/SafeFile.cpp
+@@ -189,7 +189,8 @@ float CFileDataIO::ReadFloat() const
+ {
+ 	float retVal;
+ 	Read(&retVal, sizeof(float));
+-	return retVal;
++	const auto toswap = ENDIAN_SWAP_32(reinterpret_cast<const uint32_t&>(retVal));
```

Comment:
This looks like a strict aliasing violation. You either need to memcpy or with C++20, you can do std::bitcast.

Reply:
Oops, thanks, fixed.